### ISSUE TITLE
(es-module) TypeScript found errors at build time

### ISSUE THREAD

**Issue by niquenen** [Borewit/strtok3#764]
## The problem

I used this library through the package [`file-type`](https://www.npmjs.com/package/file-type) and a build error occurred with the version [`7.0.0-alpha.8`](https://github.com/sindresorhus/file-type/blob/99cb0198bf3e55cfeda0c407607dddbab8875d91/package.json#L200). I use the module resolution set to `node16` with TypeScript. I used TypeScript `4.7.3` and Node.js `18.3.0`.

```typescript
node_modules/strtok3/lib/AbstractTokenizer.d.ts:1:58 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types.js'?

1 import { ITokenizer, IFileInfo, IReadChunkOptions } from './types';
                                                         ~~~~~~~~~
```

https://github.com/Borewit/strtok3/blob/bf6418a2ca1e8bdae1dd062b5305467930a0b2bc/lib/AbstractTokenizer.ts#L1

```typescript
node_modules/strtok3/lib/BufferTokenizer.d.ts:1:46 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './types.js'?

1 import { IFileInfo, IReadChunkOptions } from './types';
                                             ~~~~~~~~~
```

https://github.com/Borewit/strtok3/blob/bf6418a2ca1e8bdae1dd062b5305467930a0b2bc/lib/BufferTokenizer.ts#L1

To temporarily avoid this problem, the setting [`skipLibCheck`](https://github.com/Borewit/strtok3/blob/bf6418a2ca1e8bdae1dd062b5305467930a0b2bc/lib/BufferTokenizer.ts#L1) can be set to true or the [declaration files can be overridden](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#supporting-lib-from-node_modules). There are no build error with [`import type`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) and the file extension `.js`.

## Reproduction steps

1. init the project

```sh
npm init -y
```

2. open the file `package.json` and append [`{"type": "module"}`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#type-in-package-json-and-new-extensions)
3. install dependencies

```sh
npm install --save strtok3@7.0.0-alpha.8
```

4. install development dependencies

```sh
npm install --save-dev @types/node typescript@4.7.3
```

5. init TypeScript

```sh
npx tsc --init
```

6. open the file `tsconfig.json` and set `"target"` to `"es2021"`, `"module"` to `"node16"` and `"skipLibCheck"` to `"false"`

7. create a file `index.ts` and paste the script

```typescript
import { fromFile } from 'strtok3';

try {
    const tokenizer = await fromFile("somefile.bin");
}
catch (error: unknown) {
    console.error(error);
}
```

8. run a build

```sh
npx tsc
```

**Comment by Borewit** [comment]
Thank you so much for your detailed reproduction step. Looks like I also need to from `./types.js` instead of `./types` even if the file only contains TypeScript definitions.

**Comment by Borewit** [comment]
Should be fixed in [v7.0.0-alpha.9](https://github.com/Borewit/strtok3/releases/tag/v7.0.0-alpha.9)
